Stripe/STPSource.m

```diff
@@ -289,15 +287,38 @@ - (UIImage *)templateImage {
 }
 
 - (NSString *)label {
-    if (self.type == STPSourceTypeCard
-        && self.cardDetails != nil) {
-        NSString *brand = [STPCard stringFromBrand:self.cardDetails.brand];
-        return [NSString stringWithFormat:@"%@ %@", brand, self.cardDetails.last4];;
-    }
-    else {
-        return [STPCard stringFromBrand:STPCardBrandUnknown];
+    switch (self.type) {
+        case STPSourceTypeBancontact:
+            return @"Bancontact";
+        case STPSourceTypeCard:
+            if (self.cardDetails != nil) {
+                NSString *brand = [STPCard stringFromBrand:self.cardDetails.brand];
+                return [NSString stringWithFormat:@"%@ %@", brand, self.cardDetails.last4];
+            }
+            else {
+                return [STPCard stringFromBrand:STPCardBrandUnknown];
+            }
+        case STPSourceTypeGiropay:
+            return @"Giropay";
+        case STPSourceTypeIDEAL:
+            return @"iDEAL";
+        case STPSourceTypeSEPADebit:
+            return @"SEPA Direct Debit";
+        case STPSourceTypeSofort:
+            return @"SOFORT";
+        case STPSourceTypeThreeDSecure:
+            return @"3D Secure";
```

Comment:
Did you consider including more details from the source in the label strings?

3DS is an obvious one: it's wrapping a card, and the customer is probably more likely to recognize the card that's used in the Source than the fact that it was charged through 3DS. It'd also be hard for them to disambiguate if they had more than one.

I suspect the `STPSourceParam` factory methods are good references for what fields might be interesting on a per-SourceType basis.

Reply:
Yeah I did want to include more information if available but it looks like we don't make API Version guarantees on what information is included in the source specific blocks. If I'm wrong about the guarantees, then yeah we should add them appropriately.